app/src/lib/stores/pull-request-coordinator.ts

```diff
@@ -300,33 +256,31 @@ export class PullRequestCoordinator {
 /**
  * Finds local repositories related to a GitHubRepository
  *
- * * Related repos include:
- *  * **matches** — the corresponding GitHub repo (the `origin` remote for
- *    the `Repository`)
- *  * **forks** — the parent GitHub repo, if the `Repository` has one (the
+ * * Related repos include the corresponding GitHub repo (the `origin` remote for
+ *    the `Repository`) or the parent GitHub repo, if the `Repository` has one (the
  *    `upstream` remote for the `Repository`)
  *
  * @param gitHubRepository
  * @param repositories list of repositories to search for a match
- * @returns two lists of repositories: **matches** and **forks**
+ * @returns the list of repositories.
  */
 function findRepositoriesForGitHubRepository(
   gitHubRepository: GitHubRepository,
   repositories: ReadonlyArray<RepositoryWithGitHubRepository>
-) {
+): Array<RepositoryWithGitHubRepository> {
   const { dbID } = gitHubRepository
-  const matches = new Array<RepositoryWithGitHubRepository>(),
-    forks = new Array<RepositoryWithGitHubRepository>()
+  const matches = new Array<RepositoryWithGitHubRepository>()
+
   for (const r of repositories) {
     if (r.gitHubRepository.dbID === dbID) {
       matches.push(r)
     } else if (
       r.gitHubRepository.parent !== null &&
       r.gitHubRepository.parent.dbID === dbID
     ) {
-      forks.push(r)
+      matches.push(r)
     }
   }
 
-  return { matches, forks }
+  return matches
```

Comment:
Oh yes! good point!

Reply:
I've thought a bit about this and indeed I think that the current code will cause issues with grandforks, but I think that it should be changed to:

```js
return repositories.filter(
  repository => getNonForkGitHubRepository(repository).dbID === dbID
)
```

The reasoning is that this method is used to map ` GitHubRepository` object to any `Repository`  

used to calculate which `Repository` objects need to be updated after receiving an updated list of PRs. Since we fetch the list of PRs of a `Repository` by getting its `GitHubRepository` via `getNonForkGitHubRepository()`, we need to use the same method when trying to map back a `GitHubRepository` to the potential list of `Repository` objects. Does this make sense?

----

However, this raises a different question about which PRs to show when opening a grandfork. Currently if we have a `repo`, a `fork` and a `grandfork`, when opening `grandfork` on Desktop the PRs from `fork` will be shown.

I feel that it would make more sense to show the PRs from `repo`, but I don't have strong opinions about it. @billygriffin do you have any thoughts about this? (if we decide to change this we can do it in a separate PR).

Reply:
> used to calculate which `Repository` objects need to be updated after receiving an updated list of PRs. Since we fetch the list of PRs of a `Repository` by getting its `GitHubRepository` via `getNonForkGitHubRepository()`, we need to use the same method when trying to map back a `GitHubRepository` to the potential list of `Repository` objects. Does this make sense?

@rafeca oh yes, great point! i agree that this is the right solution for this method
> ```js 
> return repositories.filter(
>   repository => getNonForkGitHubRepository(repository).dbID === dbID
> )
> ```

thus far, we've explicitly tried to avoid allowing any repo to cause parent traversals more than one level (ie no repo should never have info/data from a repo that is a grandparent or higher), but also i don't think grandforks are a common enough use case to worry about beyond "will the app crash?". this logic maintains that line which i think is enough for now, but happy to reconsider more if you or @billygriffin have more thoughts!